src/scarg.lua

```diff
@@ -2,21 +2,178 @@
 -- SPDX-License-Identifier: BSD-2-Clause
 --
 -- Copyright (c) 2023 Warner Losh <imp@bsdimp.com>
+-- Copyright (c) 2024 Tyler Baxter <agge@FreeBSD.org>
 --
 
+-- Derived in large part from makesyscalls.lua:
+--
+-- SPDX-License-Identifier: BSD-2-Clause-FreeBSD
+--
+-- Copyright (c) 2019 Kyle Evans <kevans@FreeBSD.org>
+
+-- class scarg provides an interface for syscall arguments, in the parsing state.
+
+local config = require("config")
+local util = require("util")
+
 local scarg = {}
 
 scarg.__index = scarg
 
+-- xxx going to put this here for now, to get things working. going to
+-- transition to proper config merging.
+
+-- Default configuration; any of these may get replaced by a configuration file
+-- optionally specified.
+local default = {
+	abi_intptr_t = "intptr_t",
+	abi_size_t = "size_t",
+	abi_u_long = "u_long",
+	abi_long = "long",
+	abi_semid_t = "semid_t",
+	abi_ptr_array_t = "",
+	ptr_intptr_t_cast = "intptr_t",
+    -- NOTE: putting these here temporarily, in case they're needed
+    abi_flags_mask = 0,
+    abi_flags = "",
+}
+
+local function check_abi_changes(arg)
+	for k, v in pairs(config.known_abi_flags) do
```

Comment:
Changes when? I know this code is copied from makesyscalls, but it's when this ABI changes from the native ABI. A comment above the function would be good.